### DISCUSSION TITLE
Unable to specify array option with optional argument

### DISCUSSION THREAD

**Discussion by rcdailey** [spectreconsole/spectre.console#1078] [Q&A] [answered]
I have the following:

```cs
internal class Settings : ServiceCommandSettings
{
    [CommandArgument(0, "[service]")]
    [Description("One of either sonarr or radarr. If not specified, defaults to all services.")]
    public SupportedServices? Service { get; [UsedImplicitly] init; }

    [CommandOption("-c|--configs")]
    [Description("One or more YAML configuration files to load & use")]
    [TypeConverter(typeof(FileInfoConverter))]
    public IFileInfo[] Configs { get; [UsedImplicitly] init; } = Array.Empty<IFileInfo>();
}
```

The goal is to specify my command like so:

```
myapp.exe --configs file1 file2
```

However, `file2` gets treated as the value for `[service]`, not as part of the `--configs` option. Am I going about this wrong or is this a lacking feature in the library?

**Comment by nils-a** [comment, marked as answer]
You can not use space as a separator for multiple values in this way.

Either `--configs="file1 file2"` or `--configs file1,file2` should work.

(If you choose a new separator instead of space, you'd probably need to change your converter accordingly.)

**Comment by 0xced** [comment]
Would't `-c file1 -c file2` work as expected, i.e. filling the `Configs` array with the two files, without having to worry about a separator in the `FileInfoConverter`?

**Reply by nils-a** [reply]
Yes. That would work, too.

**Reply by rcdailey** [reply]
Yeah, this is the approach I'm going to go for. It doesn't require any behavior customization which I like. Thanks for the advice.
